src/main/java/org/openmrs/isanteplus/performancedata/generator/service/GeneratorService.java

```diff
@@ -0,0 +1,33 @@
+package org.openmrs.isanteplus.performancedata.generator.service;
+
+import org.openmrs.isanteplus.performancedata.generator.service.impl.PatientGeneratorService;
+import org.openmrs.isanteplus.performancedata.model.AbstractEntity;
+import org.openmrs.isanteplus.performancedata.model.connection.Inserter;
+import org.openmrs.isanteplus.performancedata.options.GeneratorOptions;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.sql.SQLException;
+import java.util.Set;
+
+@Service
+public class GeneratorService {
+
+    @Inject
+    private PatientGeneratorService patientService;
+
+    public void genereteDatabase(GeneratorOptions options) throws SQLException {
+
+        Inserter ins = new Inserter(options);
+        ins.connect();
+
+        try {
+            Set<AbstractEntity> patients = patientService.generateEntities(options.getPatients(),
```

Comment:
This has to be done in a better way obviously - you will get out of memory here when generating millions of rows